### ISSUE TITLE
cannot compress timestamps

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/pynwb#789]
## 1) Bug

pynwb throws an error when I try to compress `TimeSeries.timestamps`.

### Steps to Reproduce
```python
from pynwb import NWBHDF5IO, NWBFile, TimeSeries
from pynwb.form.backends.hdf5.h5_utils import H5DataIO
from datetime import datetime
import numpy as np

nwbfile = NWBFile('a', 'b', datetime.now().astimezone())

ts = TimeSeries('ts_name', [1, 2, 3], 'A', timestamps=H5DataIO(np.array([1., 2., 3.]), compression='gzip'))

nwbfile.add_acquisition(ts)

with NWBHDF5IO('test_out.nwb', 'w') as io:
    io.write(nwbfile)
```
Traceback:
```python
Traceback (most recent call last):
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 863, in __list_fill__
    dset = parent.create_dataset(name, shape=data_shape, dtype=dtype, **io_settings)
  File "/Users/bendichter/anaconda3/envs/dev_pynwb/lib/python3.6/site-packages/h5py/_hl/group.py", line 106, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
  File "/Users/bendichter/anaconda3/envs/dev_pynwb/lib/python3.6/site-packages/h5py/_hl/dataset.py", line 100, in make_new_dset
    tid = h5t.py_create(dtype, logical=1)
  File "h5py/h5t.pyx", line 1530, in h5py.h5t.py_create
  File "h5py/h5t.pyx", line 1552, in h5py.h5t.py_create
  File "h5py/h5t.pyx", line 1607, in h5py.h5t.py_create
TypeError: Object dtype dtype('O') has no native HDF5 equivalent

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/bendichter/dev/pynwb/scratch.py", line 14, in <module>
    io.write(nwbfile)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 381, in func_call
    return func(self, **parsed['args'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 195, in write
    call_docval_func(super(HDF5IO, self).write, kwargs)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 274, in call_docval_func
    return func(*fargs, **fkwargs)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 381, in func_call
    return func(self, **parsed['args'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/io.py", line 40, in write
    self.write_builder(f_builder, **kwargs)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 381, in func_call
    return func(self, **parsed['args'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 407, in write_builder
    self.write_group(self.__file, gbldr)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 381, in func_call
    return func(self, **parsed['args'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 550, in write_group
    self.write_group(group, sub_builder)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 381, in func_call
    return func(self, **parsed['args'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 555, in write_group
    self.write_dataset(group, sub_builder)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 381, in func_call
    return func(self, **parsed['args'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 744, in write_dataset
    dset = self.__list_fill__(parent, name, data, options)
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/backends/hdf5/h5tools.py", line 866, in __list_fill__
    raise_from(Exception(msg), exc)
  File "<string>", line 3, in raise_from
Exception: Could not create dataset timestamps in /acquisition/ts_name. Object dtype dtype('O') has no native HDF5 equivalent
```

## Checklist

- [x] Have you ensured the feature or change was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues/) ?
- [x] Have you included a brief and descriptive title?
- [x] Have you included a clear description of the problem you are trying to solve?
- [x] Have you included a minimal code snippet that reproduces the issue you are encountering?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?

**Comment by oruebel** [comment]
I have not quite found the problem yet, but it looks like the data type is set wrong to "dtype=<class 'numpy.ndarray'>" instead of the dtype of the ndarray. I believe the logic for determining data types has mostly moved out of HDF5IO up to the ObjectMapper. My guess is something may be going wrong there. @ajtritt

**Comment by oruebel** [comment]
I'm surprised this did not cause the tests or galleries to fail, i.e., I would have expected this bug to trigger any time H5DataIO was used. I've committed a fix in  #790 it's basically just two lines of code in ObjectMapper.convert_dtype to deal with DataIO objects.
